src/main/java/org/skypro/skyshop/model/controller/ShopControllerAdvice.java

```diff
@@ -0,0 +1,17 @@
+package org.skypro.skyshop.model.controller;
+
+import org.skypro.skyshop.model.ErrorHandling.ShopError;
+import org.skypro.skyshop.model.Exceptions.NoSuchProductException;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class ShopControllerAdvice {
+    @ExceptionHandler(NoSuchProductException.class)
+    public ResponseEntity<ShopError> NoSuchProductExcrption(NoSuchProductException e) {
```

Comment:
если исключение в параметре метода, то в аннотации его можно не указывать, и наоборот
```suggestion
    @ExceptionHandler
    public ResponseEntity<ShopError> NoSuchProductExcrption(NoSuchProductException e) {
```